src/empty_directory.rs

```diff
@@ -0,0 +1,31 @@
+// http://rosettacode.org/wiki/Empty_directory
+use std::fs;
+use std::path::Path;
+use std::env::args;
+
+fn main() {
+
+    // Open path
+    let path = if let Some(arg) = args().nth(1) {
+        arg
+    } else {
+        println!("You must give a path as argument");
+        return;
+    };
+
+    let dir = Path::new(&path);
+
+    let paths = match fs::read_dir(dir) {
+        Err(e) => {
+            println!("Error: {}", e);
+            return;
+        },
+        Ok(f) => f,
+    };
+
+    if paths.count() > 0 {
+        println!("Directory is not empty");
+    } else {
+        println!("Directory is empty");
+    }
```

Comment:
Hello and thanks for contributing to Rust rosetta! I have some suggestions you may find interesting :)

What do you think of the code below as a replacement of lines 16-30? You will need to remove `use std::path::Path` on line 2 (otherwise you'll get a warning).

```
match fs::read_dir(dir).map(|paths| paths.count()) {
    Err(e) => println!("Error: {}", e),
    Ok(0) => println!("Directory is empty"),
    Ok(_) => println!("Directory is not empty")
}
```

If you would like me to explain anything in the code, don't hesitate to ask!